src/time_integration/methods_2N.jl

```diff
@@ -156,6 +156,12 @@ function solve!(integrator::SimpleIntegrator2N)
         end
       end
     end
+
+    # respect maximum number of iterations
+    if integrator.iter >= integrator.opts.maxiters
```

Comment:
```suggestion
    if integrator.iter >= integrator.opts.maxiters && !integrator.finalstep
```
To prevent warnings if this would've been the last step anyways.